src/site/content/en/blog/read-files/index.md

```diff
@@ -0,0 +1,253 @@
+
+```html
+<input type="file" id="input"/>
+<p id="output"></p>
+<script>
+  const output = document.getElementById('output');
+  if (window.FileList && window.File && window.FileReader) {
+    document.getElementById('input').addEventListener('change', event => {
+      output.textContent = 'Progress: 0%';
+      const file = event.target.files[0];
+      const reader = new FileReader();
+      // `event` is a `ProgressEvent`.
+      reader.addEventListener('progress', event => {
+        // `loaded` and `total` are `ProgressEvent` properties.
+        if (event.loaded && event.total) {
+          const status = Math.round((event.loaded / event.total) * 100);
+          output.textContent = `Progress: ${status}%`;
+        }
+      });
+      reader.readAsDataURL(file);
+    }); 
+  }
+</script>
+```
+
+This example lets the user select a single file using their operating system's
+built-in file selection UI and then displays a progress bar that goes from `0%`
+to `100%` while the file's content is being loaded into memory.
+
+This is mostly the same as the [Read a file's content](#read) sample code.
+The key APIs here are the `progress` event and the 
+[`ProgressEvent`](https://developer.mozilla.org/docs/Web/API/ProgressEvent)
+object, which lets you quantify the progress of the file read operation by
+dividing its `loaded` property by its `total` property.
+
+<div class="glitch-embed-wrap" style="height: 480px; width: 100%;">
+  <iframe src="https://glitch.com/embed/#!/embed/monitor-file-read-progress?previewSize=100"
+          alt="How to monitor the progress of a file read in JavaScript."
+          style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+Browser compatibility data for the APIs used in this section:
+[`<input type="file">`](https://developer.mozilla.org/docs/Web/HTML/Element/input/file#Browser_compatibility),
+[`File`](https://developer.mozilla.org/docs/Web/API/File#Browser_compatibility),
+[`FileList`](https://developer.mozilla.org/docs/Web/API/FileList#Browser_compatibility),
+[`FileReader`](https://developer.mozilla.org/docs/Web/API/FileReader#Browser_compatibility),
+[`ProgressEvent`](https://developer.mozilla.org/docs/Web/API/ProgressEvent#Browser_compatibility)
+
+[compat]: https://developer.mozilla.org/docs/Web/API/File#Browser_compatibility
+[file]: https://developer.mozilla.org/docs/Web/API/File
+[filereader]: https://developer.mozilla.org/docs/Web/API/FileReader
+[filelist]: https://developer.mozilla.org/docs/Web/API/FileList
+[input]: https://developer.mozilla.org/docs/Web/HTML/Element/input/file
+[detection]: https://developer.mozilla.org/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection
+[events]: https://developers.google.com/analytics/devguides/collection/analyticsjs/events
+[buffer]: https://developer.mozilla.org/docs/Web/API/FileReader/readAsArrayBuffer
+[binary]: https://developer.mozilla.org/docs/Web/API/FileReader/readAsBinaryString
```

Comment:
you're not actually referencing this anywhere?

Reply:
Right, I omitted it because MDN says that it was once deprecated and then put back for backwards compatibility and recommends using an array buffer instead: https://developer.mozilla.org/en-US/docs/Web/API/FileReader/readAsBinaryString